### PR TITLE
Make EhrData subclass AnnData

### PR DESCRIPTION
**Pull request by flying-sheep**

### CODE REVIEW

File: src/ehrdata/core/ehrdata.py [hunk 1]
```diff
@@ -1,240 +1,129 @@
...
-            self._adata = adata
-
-        else:
-            raise ValueError("adata must be an AnnData object")
+        super().__init__(X=X, **kwargs)
 
         if r is not None:
+            if (r2 := self.layers.get(R_LAYER_KEY)) is not None and r2 is not r:
+                msg = "`r` is both specified and already present in `adata.layers`."
+                raise ValueError(msg)
```
Comment (flying-sheep) [inline src/ehrdata/core/ehrdata.py:34]: this is a bugfix, I’d say

File: src/ehrdata/core/ehrdata.py [hunk 2]
```diff
@@ -1,240 +1,129 @@
...
+        else:
+            self.layers[R_LAYER_KEY] = input
+
+    @r.deleter
+    def r(self) -> None:
+        del self.layers[R_LAYER_KEY]
 
     @property
-    def t(self):
+    def t(self) -> pd.DataFrame | None:
```
Comment (flying-sheep) [inline src/ehrdata/core/ehrdata.py:88]: this property isn’t really necessary right now, maybe just replace `_t: pd.DataFrame` above with `t: pd.DataFrame`.

**Review by flying-sheep — COMMENTED** [review]
Some of these should be cherry-picked even if we don’t merge this yet

File: src/ehrdata/core/ehrdata.py [hunk 3]
```diff
@@ -1,240 +1,129 @@
...
-            if isinstance(t, pd.DataFrame):
-                self.t = t
-            else:
-                raise ValueError("t must be a pandas.DataFrame")
-
+        if t is None:
+            l = 1 if r is None or len(r.shape) <= 2 else r.shape[2]
+            self.t = pd.DataFrame(index=pd.RangeIndex(l))
+        elif isinstance(t, pd.DataFrame):
+            self.t = t
```
[the remark this answers is not in the document]
Comment (eroell) [reply src/ehrdata/core/ehrdata.py:57]: looks good! it indeed does.

File: src/ehrdata/core/ehrdata.py (hunk 2 above)
Comment (eroell) [reply src/ehrdata/core/ehrdata.py:124]: Yeah hahah true I think this was one of the first attributes I've made and considered so ancient I didnt dare to switch during hackathon time.
If you see it too, I do now

File: src/ehrdata/core/ehrdata.py [hunk 4]
```diff
@@ -1,240 +1,129 @@
...
-        obsm=None,
-        varm=None,
-        layers=None,
-        raw=None,
-        dtype=None,
-        shape=None,
-        filename=None,
-        filemode=None,
-        asview=None,
+        r: np.ndarray | None = None,
```
Comment (eroell) [inline src/ehrdata/core/ehrdata.py:38]: Waaay cleaner. Awesome.

File: src/ehrdata/core/ehrdata.py (hunk 1 above)
Comment (eroell) [reply src/ehrdata/core/ehrdata.py:48]: the behavior before would have just overwritten the field I suppose with the r argument?
in any case I did not test the behaviour in this edge case of a properly named layer AND the r handed over too, and this should also not be allowed. good.

**Review by eroell — APPROVED** [review]
thanks so much!
